lib/node_modules/@stdlib/math/base/special/hyp2f1/src/main.c

```diff
@@ -0,0 +1,598 @@
+	isIntD = isInteger( d );
+	t1 = stdlib_base_abs( b-a );
+
+	if ( stdlib_base_is_nan( a ) || stdlib_base_is_nan( b ) || stdlib_base_is_nan( c ) || stdlib_base_is_nan( x ) ) {
+		return STDLIB_CONSTANT_FLOAT64_NAN;
+	}
+
+	if ( x == 0.0 ) {
+		return 1.0;
+	}
+
+	if ( ( a == 0.0 || b == 0.0 ) && c != 0.0 ) {
+		return 1.0;
+	}
+
+	// The transformation for c - a or c - b negative integer (AMS55 #15.3.3)...
+	if ( d <= -1.0 && !( !isIntD && s < 0.0 ) && !( negIntA || negIntB ) ) {
+		return stdlib_base_pow( s, d ) * stdlib_base_hyp2f1( c-a, c-b, c, x );
+	}
+
+	// Check whether the series diverges...
+	if ( d <= 0.0 && x == 1.0 && !( negIntA || negIntB ) ) {
+		return STDLIB_CONSTANT_FLOAT64_PINF;
+	}
+
+	if ( ax < 1.0 || x == -1.0 ) {
+		if ( b == c ) {
+			// 2F1(a,b;b;x) = (1-x)**(-a):
+			y = stdlib_base_pow( s, -a );
+			return y;
+		}
+		if ( a == c ) {
+			// 2F1(a,b;a;x) = (1-x)**(-b):
+			y = stdlib_base_pow( s, -b );
+			return y;
+		}
+	}
+
+	if ( negIntC ) {
+		// Check if termination before explosion...
+		if ( negIntA && ( ia > ic ) ) {
+			y = hyt2f1( a, b, c, x, &err );
+			return y;
+		}
+		if ( negIntB && ( ib > ic ) ) {
+			y = hyt2f1( a, b, c, x, &err );
+			return y;
+		}
+		return STDLIB_CONSTANT_FLOAT64_PINF;
+	}
+
+	// Check whether the function is a polynomial before trying the power series expansion...
+	if ( negIntA || negIntB )	{
+		y = hyt2f1( a, b, c, x, &err );
+		return y;
+	}
+
+	// The following transform has a pole for integer b - a and may cause large cancellation errors near |1/x| = 1 (AMS55 #15.3.7)...
+	if ( x < -2.0 && !isInteger( t1 ) ) {
+		p = stdlib_base_hyp2f1( a, 1.0-c+a, 1.0-b+a, 1.0/x );
+		q = stdlib_base_hyp2f1( b, 1.0-c+b, 1.0-a+b, 1.0/x );
+		p *= stdlib_base_pow( -x, -a );
+		q *= stdlib_base_pow( -x, -b );
+		t1 = stdlib_base_gamma( c );
+		s = t1 * stdlib_base_gamma( b-a ) / ( stdlib_base_gamma( b ) * stdlib_base_gamma( c-a ) );
+		y = t1 * stdlib_base_gamma( a-b ) / ( stdlib_base_gamma( a ) * stdlib_base_gamma( c-b ) );
+		return ( s*p ) + ( y*q );
+	}
+	if ( x < -1.0 ) {
+		// Transformation based on AMS55 #15.3.4...
+		if ( stdlib_base_abs( a ) < stdlib_base_abs( b ) ) {
+			return stdlib_base_pow( s, -a ) * stdlib_base_hyp2f1( a, c-b, c, x/( x-1.0 ) );
+		}
+		// Transformation based on AMS55 #15.3.5:
+		return stdlib_base_pow( s, -b ) * stdlib_base_hyp2f1( b, c-a, c, x/( x-1.0 ) );
+	}
+
+	// The series diverges for `|x|` greater than unity if above checks fail...
+	if ( ax > 1.0 ) {
+		return STDLIB_CONSTANT_FLOAT64_PINF;
+	}
+
+	p = c - a;
+	r = c - b;
+	negIntCaOrCb = isNonPositiveInteger( p ) || isNonPositiveInteger( r );
+
+	// If `|x|` is equal to unity, the function is a polynomial...
+	if ( ax == 1.0 ) {
+		if ( x > 0.0 ) {
+			if ( negIntCaOrCb ) {
+				if ( d >= 0.0 ) {
+					// Transformation based on AMS55 #15.3.3:
+					y = stdlib_base_pow( s, d ) * hys2f1( c - a, c - b, c, x, &err );
+					return y;
+				} else {
+					return STDLIB_CONSTANT_FLOAT64_PINF;
+				}
```

Comment:
```suggestion
				}
				return STDLIB_CONSTANT_FLOAT64_PINF;
```

Else block is not necessary.